src/main/kotlin/com/vk/kphpstorm/intentions/AddImmutableClassAnnotationIntention.kt

```diff
@@ -0,0 +1,86 @@
+package com.vk.kphpstorm.intentions
+
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.codeInspection.util.IntentionFamilyName
+import com.intellij.codeInspection.util.IntentionName
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.search.LocalSearchScope
+import com.intellij.psi.search.searches.ReferencesSearch
+import com.intellij.psi.util.findParentInFile
+import com.jetbrains.php.lang.psi.elements.AssignmentExpression
+import com.jetbrains.php.lang.psi.elements.PhpClass
+import com.vk.kphpstorm.inspections.helpers.PhpDocPsiBuilder
+import com.vk.kphpstorm.kphptags.KphpImmutableClassDocTag
+
+class AddImmutableClassAnnotationIntention : PsiElementBaseIntentionAction() {
+    override fun getText(): @IntentionName String = "Add @kphp-immutable-class"
+    override fun getFamilyName(): @IntentionFamilyName String = getText()
+
+    override fun isAvailable(
+        project: Project,
+        editor: Editor?,
+        element: PsiElement
+    ): Boolean {
+        if (!element.isClassNameNode()) {
+            return false
+        }
+
+        val klass = element.parent as PhpClass
+        val klassDocNode = klass.docComment
+
+        // do not suggest if already present
+        if (klassDocNode != null && KphpImmutableClassDocTag.existsInDocComment(klassDocNode)) {
+            return false
+        }
+
+        return !isClassLocallyImmutable(klass)
+    }
+
+    /**
+     * Simple local class mutability check. If there is any field mutation in class,
+     * the class is mutable. The only exception is the class constructor
+     */
+    private fun isClassLocallyImmutable(klass: PhpClass): Boolean {
+        val searchScope = LocalSearchScope(klass)
+        for (field in klass.fields) {
+            val hasAnyMutation = ReferencesSearch.search(field, searchScope).any { ref ->
+                val element = ref.element
+
+                isMutatingOp(element) && !isInClassConstructor(klass, element)
+            }
+
+            if (hasAnyMutation) {
+                return true
+            }
+        }
+
+        return false
+    }
+
+    private fun isMutatingOp(psiElement: PsiElement): Boolean {
+        val parent = psiElement.parent
+
+        return parent is AssignmentExpression
```

Comment:
You need to add a check for a lhs here

In the following code I should be able to use the intention:
<img width="358" height="254" alt="Image" src="https://github.com/user-attachments/assets/32b488a3-078a-460b-99f1-eb2f5588a243" />